From 594c0acb735e6351621cff3e9fb76b6fd515819e Mon Sep 17 00:00:00 2001
From: gollarx <ratnakumarix.golla@intel.com>
Date: Mon, 14 Nov 2022 14:56:32 +0530
Subject: [PATCH] Fix for VTS MultiRelease & Timeout encoder TCs

Vts MultiRelease TC's and Timout encoder TC's are failing due to
modifiable variable(which is 0) is resulting the failure in DoConfig Method in
encoder component.

Tracked-On: OAM-104668
Signed-off-by: gollarx <ratnakumarix.golla@intel.com>
---
 c2_components/src/mfx_c2_encoder_component.cpp | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/c2_components/src/mfx_c2_encoder_component.cpp b/c2_components/src/mfx_c2_encoder_component.cpp
index 64905e3..9f13aac 100755
--- a/c2_components/src/mfx_c2_encoder_component.cpp
+++ b/c2_components/src/mfx_c2_encoder_component.cpp
@@ -1682,11 +1682,9 @@ void MfxC2EncoderComponent::DoConfig(const std::vector<C2Param*> &params,
             continue;
         }
         // check whether plugin is in a correct state to apply this parameter
-        // the check is bypassed for bitrate parameter as it should be updatable
-        // despite its type 'info' (workaround of Google's bug)
         bool modifiable = (param->coreIndex().coreIndex() == kParamIndexBitrate) ||
-            (param->kind() == C2Param::TUNING) || (param->kind() == C2Param::INFO) ||
-            (param->kind() == C2Param::SETTING && m_state == State::STOPPED);
+		((param->kind() & C2Param::TUNING) != 0) ||
+		m_state == State::STOPPED; /* all kinds, even INFO might be set by stagefright */
 
         if (!modifiable) {
             failures->push_back(MakeC2SettingResult(C2ParamField(param), C2SettingResult::READ_ONLY));
-- 
2.38.1

